Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugByteReporter.cpp

```diff
@@ -0,0 +1,203 @@
+    }
+
+    size_t MultiplayerDebugByteReporter::GetMaxBytes() const
+    {
+        return m_maxBytes;
+    }
+
+    size_t MultiplayerDebugByteReporter::GetMinBytes() const
+    {
+        return m_minBytes;
+    }
+
+    size_t MultiplayerDebugByteReporter::GetTotalBytes() const
+    {
+        return m_totalBytes;
+    }
+
+    float MultiplayerDebugByteReporter::GetKbitsPerSecond()
+    {
+        const auto now = AZStd::chrono::monotonic_clock::now();
+
+        // Check the amount of time elapsed and update totals if necessary.
+        // Time here is measured in whole seconds from the epoch, providing synchronization in
+        // reporting intervals across all byte reporters.
+        const AZStd::chrono::seconds nowSeconds = AZStd::chrono::duration_cast<AZStd::chrono::seconds>(now.time_since_epoch());
+        const AZStd::chrono::seconds secondsSinceLastUpdate = nowSeconds -
+            AZStd::chrono::duration_cast<AZStd::chrono::seconds>(m_lastUpdateTime.time_since_epoch());
+        if (secondsSinceLastUpdate.count())
+        {
+            // normalize over elapsed milliseconds
+            constexpr int k_millisecondsPerSecond = 1000;
+            const auto msSinceLastUpdate = AZStd::chrono::duration_cast<AZStd::chrono::milliseconds>(now - m_lastUpdateTime);
+            m_totalBytesLastSecond = k_millisecondsPerSecond * aznumeric_cast<float>(m_totalBytesThisSecond) / aznumeric_cast<float>(msSinceLastUpdate.count());
+            m_totalBytesThisSecond = 0;
+            m_lastUpdateTime = now;
+        }
+
+        constexpr float bitsPerByte = 8.0f;
+        constexpr int bitsPerKilobit = 1024;
+        return bitsPerByte * m_totalBytesLastSecond / bitsPerKilobit;
+    }
+
+    void MultiplayerDebugByteReporter::Combine(const MultiplayerDebugByteReporter& other)
+    {
+        m_count += other.m_count;
+        m_totalBytes += other.m_totalBytes;
+        m_totalBytesThisSecond += other.m_totalBytesThisSecond;
+        m_minBytes = AZStd::GetMin(m_minBytes, other.m_minBytes);
+        m_maxBytes = AZStd::GetMax(m_maxBytes, other.m_maxBytes);
+    }
+
+    void MultiplayerDebugByteReporter::Reset()
+    {
+        m_count = 0;
+        m_totalBytes = 0;
+        m_totalBytesThisSecond = 0;
+        m_totalBytesLastSecond = 0;
+        m_minBytes = std::numeric_limits<decltype(m_minBytes)>::max();
+        m_maxBytes = 0;
+        m_aggregateBytes = 0;
+    }
+
+    void MultiplayerDebugComponentReporter::ReportField(const char* fieldName, size_t byteSize)
+    {
+        MultiplayerDebugByteReporter::AggregateBytes(byteSize);
+        m_fieldReports[fieldName].ReportBytes(byteSize);
+    }
+
+    void MultiplayerDebugComponentReporter::ReportFragmentEnd()
+    {
+        MultiplayerDebugByteReporter::ReportAggregateBytes();
+        m_componentDirtyBytes.ReportAggregateBytes();
+    }
+
+    AZStd::vector<MultiplayerDebugComponentReporter::Report> MultiplayerDebugComponentReporter::GetFieldReports()
```

Comment:
Just to check, this is intentionally not const due to the ByteReporter pointer in the Report type?

Reply:
Yeah, it's the intentional design here. Same as GetComponentReports()